rust/tests/integration/plain_transport.rs

```diff
@@ -403,6 +412,105 @@ fn create_non_bindable_ip() {
                     ip: "8.8.8.8".parse().unwrap(),
                     announced_ip: None,
                     port: None,
+                    flags: None,
+                    send_buffer_size: None,
+                    recv_buffer_size: None,
+                }))
+                .await,
+            Err(RequestError::Response { .. }),
+        ));
+    });
+}
+
+#[cfg(not(target_os = "windows"))]
+#[test]
+fn create_two_transports_binding_to_same_ip_port_with_udp_reuse_port_flag_succeed() {
+    future::block_on(async move {
+        let (_worker, router) = init().await;
+
+        let multicast_ip = "224.0.0.1".parse().unwrap();
+        let port = pick_unused_port().unwrap();
+
+        // Transport 1.
+        let _ = router
+            .create_plain_transport({
+                PlainTransportOptions::new(ListenInfo {
+                    protocol: Protocol::Udp,
+                    ip: multicast_ip,
+                    announced_ip: None,
+                    port: Some(port),
+                    flags: Some(SocketFlags {
+                        ipv6_only: false,
+                        udp_reuse_port: true,
```

Comment:
So the problem is that this test passes even if you set `udp_reuse_port: false` in transport 1 or transport 2 (below). In Node it behaves correctly (it fails if any of the transports has `udp_reuse_port: false`.

I've made some checks in `PlainTransport.cpp` and my conclusion is that, as far as `flags` is not None, the FBS object received by the worker has `udpReusePort` with value `true`!! I've confirmed it by adding this code in `PlainTransport` constructor:

```c++
if (this->listenInfo.flags.udpReusePort) {
	MS_DUMP("Oh, true?? ok I am converting it to false!!!");
	this->listenInfo.flags.udpReusePort = false;
}
```

In Rust, such a log ONLY shows up when a `PlainTransport` is created (by a test) with `flags` NOT set to `None`, and in that case, even if you pass ` flags: Some(SocketFlags { ipv6_only: false, udp_reuse_port: false }`, the `MS_DUMP()` is printed, meaning that received `udpReusePort` is `true`!!!

HELP.

Reply:
Wait. I may be completely wrong. I finally can print those logs. In `PlainTransport` constructor I've added:

```c++
MS_ERROR_STD("options->listenInfo()->flags()->udpReusePort(): %u", options->listenInfo()->flags()->udpReusePort());
```

And I run tests as follows:

```
DEBUG="mediasoup:ERROR:Worker" cargo test
```

and yes, I only see two lines with value true (1):

```
options->listenInfo()->flags()->udpReusePort(): 1
```

which matches the number of PlainTransports created by the tests with `udp_reuse_port: true`, so we are good here.

So then what? How is it possible that this test DOESNT FAIL if we change `udp_reuse_port` to `false`?? It perfectly fails in Node as expected. And Rust is NOT doing some magic by adding `SO_REUSEADDR` to sockets created in worker using libuv API, right? Because that's not possible, right?

Reply:
Fixed here: https://github.com/versatica/mediasoup/pull/1291/commits/7b935b61136793c0f1644ad3fcef27b65234d9c3